src/lib/storage/table.cpp

```diff
@@ -71,12 +75,15 @@ void Table::add_column(const std::string& name, DataType data_type, bool nullabl
 
 void Table::append(std::vector<AllTypeVariant> values) {
   // TODO(Anyone): Chunks should be preallocated for chunk size
-  if (_chunks.back()->size() == _max_chunk_size) create_new_chunk();
-
-  _chunks.back()->append(values);
+  auto partition_id = _partition_schema->get_matching_partition_for(values);
+  auto last_chunk = _partition_schema->last_chunk(partition_id);
+  if (last_chunk->size() >= max_chunk_size()) {
+    create_new_chunk(partition_id);
+  }
+  _partition_schema->append(values, partition_id);
```

Comment:
This block mixes up responsibilities. On the one hand, the table checks if the last chunk in the partition is full, but on the other hand, the partitioning is responsible for then finding the last chunk to append to.